### PR TITLE
msvc and borland should define ACE_HAS_WIN32_STRUCTURAL_EXCEPTIONS, r…

### PR DESCRIPTION
**Pull request by jwillemsen**
…emoves mingw check from the common header

    * ACE/ace/config-win32-borland.h:
    * ACE/ace/config-win32-common.h:
    * ACE/ace/config-win32-msvc.h:

### CODE REVIEW

File: ACE/ace/config-win32-borland.h [hunk 1]
```diff
@@ -60,9 +60,10 @@
 
 # include "ace/config-win32-common.h"
 
-# define ACE_WSTRING_HAS_USHORT_SUPPORT 1
-# define ACE_HAS_DIRENT
+#define ACE_WSTRING_HAS_USHORT_SUPPORT 1
+#define ACE_HAS_DIRENT
 
+#define ACE_HAS_WIN32_STRUCTURAL_EXCEPTIONS
```
Comment (mitza-oci) [inline ACE/ace/config-win32-borland.h:66]: "Structural Exceptions" should be "Structured Exceptions"
https://docs.microsoft.com/en-us/windows/win32/debug/structured-exception-handling

Comment (jwillemsen) [reply ACE/ace/config-win32-borland.h:66]: Yes, I know, renamjng the incorrect macro could break user code, not sure whether we should do that

Comment (mitza-oci) [reply ACE/ace/config-win32-borland.h:66]: ACE 7 may be the opportunity to change it.  Is this for the case where a user customizes `config.h` or where the user is checking the configuration in code that uses ACE?

Comment (jwillemsen) [reply ACE/ace/config-win32-borland.h:66]: I think more where the user checks this define is his own code

**Comment by jwillemsen** [comment]
See https://github.com/DOCGroup/ACE_TAO/pull/1578, old macro is defined for backwards compatibility for the moment

**Pull request merged**